tool/src/org/antlr/v4/codegen/target/GoTarget.java

```diff
@@ -54,6 +57,26 @@ protected void addBadWords() {
 	}
 
 	@Override
+	protected void genFile(Grammar g, ST outputFileST, String fileName) {
+		super.genFile(g, outputFileST, fileName);
+		gofmt(fileName); // consider commenting out when debugging
+	}
+
+	private void gofmt(String fileName) {
+		// If the generated source file is for any reason invalid, gofmt will not change the file and generate output on
+		// stderr. The output of stderr will be lost, gofmt may block asynchronously if the output buffer is not
+		// drained, but in that case gofmt will be killed as soon the JVM terminates. In normal operations the gofmt
+		// process will not get killed prematurely and not create a race condition with the lifecycle of the JVM.
```

Comment:
If we'd program in Java 1.7+ and upstream would import apache-common, we could redirect all output to /dev/null in the same line of code that runs this subcommand. Without the Redirect that 1.7 introduces and the null redirect target that apache-commons provides, we'd have to manually take the two streams and write a class that drains them in a loop, decorated with error handling. 

I am not going to do that, unless someone convinces me this code is harmful.